spinalcordtoolbox/aggregate_slicewise.py

```diff
@@ -306,7 +308,14 @@ def aggregate_per_slice_or_level(metric, mask=None, slices=[], levels=[], distan
     for slicegroup in slicegroups:
         # add distance from PMJ info
         if distance_pmj is not None:
-            agg_metric[slicegroup]['DistancePMJ'] = [distance_pmj]
+            agg_metric[slicegroup]['DistancePMJ'] = distance_pmj
+            length_pmj = None
+        elif length_pmj is not None:
+            index = np.where(length_pmj[1, :] == slicegroup)[0]
+            if index:
+                agg_metric[slicegroup]['DistancePMJ'] = round(length_pmj[0, index[0]], 2)
```

Comment:
Based on how this code is using `length_pmj`, it seems like it would be much more natural if it was a dict instead of a 2-row numpy array. Then, you could say something like:
```py3
agg_metric[slicegroup]['DistancePMJ'] = None
for slice in slicegroup:
    if slice in length_pmj:
        agg_metric[slicegroup]['DistancePMJ'] = round(length_pmj[slice], 2)
        break
```